src/Autocomplete/src/EntityAutocompleterInterface.php

```diff
@@ -18,30 +18,52 @@
 /**
  * Interface for classes that will have an "autocomplete" endpoint exposed.
  *
+ * @template T of object
+ *
+ * @method array getResults(object $entity) Returns the autocomplete result, usually the label as "text" and the value as "value".
+ *                                          May be used to expose extra options such as "disabled" so that tom-select's rendering may be customized.
  * @method mixed getGroupBy() Return group_by option.
  */
 interface EntityAutocompleterInterface
 {
     /**
      * The fully-qualified entity class this will be autocompleting.
+     *
+     * @return class-string<T>
      */
     public function getEntityClass(): string;
 
     /**
      * Create a query builder that filters for the given "query".
+     *
+     * @param EntityRepository<T> $repository
      */
     public function createFilteredQueryBuilder(EntityRepository $repository, string $query): QueryBuilder;
 
     /**
      * Returns the "choice_label" used to display this entity.
+     *
+     * @param T $entity
      */
     public function getLabel(object $entity): string;
 
     /**
      * Returns the "value" attribute for this entity, usually the id.
+     *
+     * @param T $entity
      */
     public function getValue(object $entity): mixed;
 
+    /**
+     * Returns the autocomplete result, usually the label as "text" and the value as "value".
+     * May be used to expose extra options such as "disabled" so that tom-select's rendering may be customized.
+     *
+     * @param T $entity
+     *
+     * @return array<string, mixed>
+     */
+    /* public function getResult(object $entity): array; */
+
```

Comment:
Well... i'm now pretty sure. 

We have 
* a `getLabel` method (equivalent to [choice_label](https://symfony.com/doc/current/reference/forms/types/choice.html#choice-label) in Form options 
* a `getValue` method (equivalent to [choice_value](https://symfony.com/doc/current/reference/forms/types/choice.html#choice-value) in Form options)

Instead of introducing a new method that englobes/replaces the previously existing one, i'd rather add a new method 

```php
/**
 * @return array<string, string|null>|null
 */
public function getAttributes(object $entity): ?array
```

In your scenario, you could return ['disabled' => true] when you'd want to.

Wdyt ?